lib/resty/upstream/healthcheck.lua

```diff
@@ -600,8 +638,14 @@ function _M.spawn_checker(opts)
         statuses = statuses,
         version = 0,
         concurrency = concur,
+        http_version = opts.http_version or "1.1",
+        expected_body = opts.expected_body
     }
 
+    if ctx.http_version ~= "1.1" or ctx.http_version ~= "1.0" then 
```

Comment:
I just commented on the line, but actually the checks should be done further up, before the `ctx` table is created. 

and then it should be something like;
```lua
local http_allowed = { ["1.0"] = true, ["1.1"] = true }

local http_version = opts.http_version or "1.1"
if not http_allowed[http_version] then
    return nil, "invalid http version: "..tostring(http_version)
end

````